### ISSUE TITLE
Build error in typescript

### ISSUE THREAD

**Issue by ajainarayanan** [jsplumb/jsplumb#695]
#### Issue:
- Installed jsplumb #2.5.8 and `@types/jsplumb` 2.5.7
- During compile step getting these two errors,
 - Duplicate Endpoint 
```Javascript
ERROR in <path-to-project>/node_modules/jsplumb/index.d.ts (12,6): Duplicate identifier 
'Endpoint'.
```
- Not a module,
```Javascript
File <path-to-project>/node_modules/jsplumb/index.d.ts is not a module.
```

When I see the node_modules folder structure I am seeing `index.d.ts` file at the root. Is there any specific reason why a type file was added to jsplumb npm module compared to `@types/jsplumb`? Which is the correct version to use? 

Is anybody else facing this issue and have a workaround or am I missing something in my setup?

**Comment by sporritt** [comment]
i didnt have anything to do with publishing @types/jsplumb

and it doesnt seem to do anything anyway

![image](https://user-images.githubusercontent.com/262720/33919843-e7fe682e-e00e-11e7-818d-d1055007fe2f.png)

**Comment by sporritt** [comment]
that should read "with publish at types/jsplumb"

but github seem to have a wee bug in their ui

**Comment by ajainarayanan** [comment]
@sporritt I see. Ok I have removed `@types/jsplumb` but I am still seeing the above stated two errors. My current workaround to start my ts project is to manually edit the `index.d.ts` under `node_modules/jsplumb`. Is it ok if I send in a pull request for the `Endpoint` change?

Also is there a fix for the second error from the description?

**Comment by Nightbr** [comment]
Same bug here, it seems the old typing has been removed from Global @types: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/21195

I think this PR can fix https://github.com/jsplumb/jsplumb/pull/677

Just remove this line and it should work https://github.com/jsplumb/jsplumb/blob/master/index.d.ts#L12

Any update on this? future release?

**Comment by ajainarayanan** [comment]
@Nightbr Are you getting the second error mentioned in the description? How do you use jsplumb interface(types) vs the library functions? (sorry if the question is not clear. I am relatively new to typescript).

**Comment by Nightbr** [comment]
I am not getting the second error and I manage to use jsplumb into my Angular app (typescript). 
I just import it into app.ts (or app.module.ts for Angular structure)

`import 'jsplumb';`

**Comment by ajainarayanan** [comment]
I am using jsplumb in a react app. I am sure I have my tsconfig.json mis-configured in someway. If anyone happened to resolve this issue please do update this thread. @sporritt sorry for the spam. Please close this issue if you feel its not reporting any bugs.

**Comment by Nightbr** [comment]
If it can help you:

tsconfig.app.json
```json
{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "outDir": "../out-tsc/app",
    "module": "es2015",
    "baseUrl": "",
    "types": []
  },
  "exclude": [
    "test.ts",
    "**/*.spec.ts"
  ]
}
```

typings.d.ts
```typescript
/* SystemJS module definition */
declare var System: any;
declare var module: { id: string };
declare var require: any;
declare module 'leaflet';
```

**Comment by ajainarayanan** [comment]
@Nightbr Got it to working without any errors. @sporritt since the compile step right now breaks would you be willing to accept a pull request to fix that alone? I am unable to use jsplumb in my ts project without two changes to the index.d.ts file.

I see an already existing PR on type definition file just wanted to confirm before creating a pull request.

**Comment by sporritt** [comment]
@ajainarayanan sure, that'd be good.

**Comment by ajainarayanan** [comment]
@sporritt Have created a small PR https://github.com/jsplumb/jsplumb/pull/699. Let me know if I missed something.

**Comment by ajainarayanan** [comment]
@sporritt thanks for the quick response! is there a tentative time planned for 2.5.12 version of jsplumb?

**Comment by sporritt** [comment]
i released 2.5.12 today.
